packages/core/src/render3/instructions.ts

```diff
@@ -1387,14 +1387,18 @@ export function bind2(prefix: string, v0: any, i0: string, v1: any, suffix: stri
     if (typeof currentView.bindingStartIndex !== 'number') {
       bindingIndex = currentView.bindingStartIndex = data.length;
     }
-    data[bindingIndex++] = {v0: v0, v1: v1};
+    data[bindingIndex++] = v0;
+    data[bindingIndex++] = v1;
   } else {
-    const parts: {v0: any, v1: any} = data[bindingIndex++];
-    if (v0 === NO_CHANGE) v0 = parts.v0;
-    if (v1 === NO_CHANGE) v1 = parts.v1;
-    if (different = (isDifferent(parts.v0, v0) || isDifferent(parts.v1, v1))) {
-      parts.v0 = v0;
-      parts.v1 = v1;
+    const part0Index = bindingIndex++;
+    const part1Index = bindingIndex++;
+    const part0 = data[part0Index];
```

Comment:
I would inline the `++` as `data[bindingIndex++]`. This saves the creation of local war. (See next comment)